src/components/organizations/CreateOrganization.tsx

```diff
@@ -1,77 +1,15 @@
 import * as React from 'react';
+import { OrgAddEdit } from './OrgAddEdit';
+import { newOrganizationFields } from './newOrganizationFields';
 import { Page } from '../page/Page';
-
-import ImageUploader from 'react-images-upload';
-
-export const fieldSetChildStyle = {
-    "display": "block",
-    "padding": "10px",
-    "margin": "10px auto 0",
-    "border-radius": "5px",
-    "border": "1px solid #a0a0a0",
-    "width": "400px",
-    "text-align": "center"
-}
-
-export const submitButtonStyle = { 
-    "color": "white",
-    "background-color": "rgba(255, 0, 0, 0.5)",
-    "font-weight": "bold",
-    "font-size": "20px",
-    "cursor": "pointer"
-}
+import { HPButton } from '../forms/HPButton';
 
 export const CreateOrganization = (props) => {
-
-    const updateNewOrgTitle = (event:React.ChangeEvent<HTMLInputElement>) => {
-        const newOrgTitle = event.target.value;
-        props.updateNewOrgTitle(newOrgTitle);
-    }
-
-    const updateNewOrgMission = (event:React.ChangeEvent<HTMLTextAreaElement>) => {
-        const newOrgMission = event.target.value;
-        props.updateNewOrgMission(newOrgMission);
-    }
-
-    const updateNewOrgUrl = (event:React.ChangeEvent<HTMLInputElement>) => {
-        const newOrgUrl = event.target.value;
-        props.updateNewOrgUrl(newOrgUrl);
-    }
-
-    const updateNewOrgLogo = async (image) => {
-        const reader = new FileReader();
-        const file = image[0]
-        reader.onload = function(e:any) {
-            if(e.target) {
-                const src = e.target.result; 
-                file.src = src;
-                props.updateNewOrgLogo(file);
-            } 
-        };
-        reader.readAsDataURL(file);
-    }
-
-    const addNewOrganization = () => {
-        props.facebookUserSession 
-            ? props.addNewOrganization(props.facebookUserSession.email) 
-            : alert("You must be logged in to create an Organization!");
-    }
-
-    //todo: ideally the state for this component is associated with the instance of it
-    //todo: force single image not working?
+    const { bindFields, addNewOrganization, facebookUserSession: { email }, } = props;
+    const fields = bindFields(newOrganizationFields())
```

Comment:
here is an example where we want to bring in the `newOrganizationFields()`, but those are not yet "bound" to the particular current state, or the ability to re-render the app. So, `newOrganizationFields()` returns a `Dictionary<FieldBinder<...>>`, something like:

```
{
  title: FieldBinder(...),
  homePage: FieldBinder(...),
  ...
}
```

and then by calling `bindFields`, those general field definitions, which are defined in a way that does not bind them to any state, or any ability to re-render the app, get "bound" to the state and re-rendering capability. At that point, `fields` now looks like a nice set of named field props, ready to pass into the child:

```
{
  title: Field(...bound to state and re-render ability),
  homePage: Field(...),
  ...
}
```

Reply:
oooo